### PR TITLE
Docs: Documentation for Mask-RCNN

### PR DESCRIPTION
**Pull request by eric-kwok-nt**

### CODE REVIEW

**Review by liyier90 — CHANGES_REQUESTED** [review]
Some additional formatting/consistency issues:
[1](https://github.com/aimakerspace/PeekingDuck/blob/node-segment/peekingduck/pipeline/nodes/model/mask_rcnn.py#L29): `"r101-fpn"`  should be monospace as well.
[2](https://github.com/aimakerspace/PeekingDuck/blob/node-segment/peekingduck/pipeline/nodes/model/mask_rcnn.py#L76): The inference code referenced here does not match the code pointed by the URL in Bibliography. We should probably avoid using the `stable` URL as it will change when torchvision increments its version number.

File: docs/source/resources/01e_instance_segmentation.rst [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
++--------------+------------+-----------+-------------------+-------------------+
+|              |            |           | CPU               | GPU               |
+|              |            |           +--------+----------+--------+----------+
+| Model        | Type       | Size      | single | multiple | single | multiple |
++==============+============+===========+========+==========+========+==========+
+|              | r50-fpn    | 800, 1333 | 0.76   | 0.72     | 22.30  | 18.58    |
++              +------------+-----------+--------+----------+--------+----------+
+| Mask R-CNN   | r101-fpn   | 800, 1333 | 0.61   | 0.57     | 17.14  | 14.83    |
++--------------+------------+-----------+--------+----------+--------+----------+
```
[the remark this answers is not in the document]
Comment (eric-kwok-nt) [reply docs/source/resources/01e_instance_segmentation.rst:32]: Changed to dash

File: docs/source/resources/01e_instance_segmentation.rst [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
++==============+===========+============+======+===================+===================+=================+==================+=================+=================+==================+===================+=================+==================+=================+
+|              | r50-fpn   | 800, 1333  | 34.5 | 56.0              | 36.7              | 17.8            | 37.9             | 47.1            | 29.7            | 45.6             | 47.6              | 27.4            | 51.5             | 63.8            |
++              +-----------+------------+------+-------------------+-------------------+-----------------+------------------+-----------------+-----------------+------------------+-------------------+-----------------+------------------+-----------------+
+| Mask R-CNN   | r101-fpn  | 800, 1333  | 36.6 | 58.4              | 39.1              | 19.0            | 40.8             | 50.1            | 31.0            | 48.0             | 50.2              | 29.7            | 55.0             | 66.6            |
```
[the remark this answers is not in the document]
Comment (eric-kwok-nt) [reply docs/source/resources/01e_instance_segmentation.rst:68]: Amended

File: docs/source/resources/01e_instance_segmentation.rst [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
++==============+===========+============+======+===================+===================+=================+==================+=================+=================+==================+===================+=================+==================+=================+
+|              | r50-fpn   | 800, 1333  | 37.6 | 59.1              | 40.9              | 21.3            | 41.3             | 49.3            | 31.3            | 49.3             | 51.6              | 32.1            | 55.6             | 66.5            |
++              +-----------+------------+------+-------------------+-------------------+-----------------+------------------+-----------------+-----------------+------------------+-------------------+-----------------+------------------+-----------------+
+| Mask R-CNN   | r101-fpn  | 800, 1333  | 41.3 | 61.7              | 44.7              | 22.8            | 45.9             | 54.4            | 33.9            | 53.5             | 56.1              | 35.0            | 61.4             | 71.4            |
```
[the remark this answers is not in the document]
Comment (eric-kwok-nt) [reply docs/source/resources/01e_instance_segmentation.rst:79]: Amended

File: docs/source/resources/02_bibliography.rst [hunk 4]
```diff
@@ -68,3 +68,12 @@ Object Tracking
...
 | 🦆 FairMOT                | `✔ <https://arxiv.org/abs/2004.01888>`__    | `✔ <https://github.com/ifzhang/FairMOT>`__                |
 +---------------------------+---------------------------------------------+-----------------------------------------------------------+
+
+Instance Segmentation
+=====================
+
++--------------------+----------------------------------------------+----------------------------------------------------------------------------------------------------------------------+
+| Reference          | Paper                                        | Code                                                                                                                 |
++====================+==============================================+======================================================================================================================+
+| 🦆 Mask R-CNN      | `✔ <https://arxiv.org/abs/1703.06870>`__     | `Torchvision Models <https://pytorch.org/vision/0.12/_modules/torchvision/models/detection/mask_rcnn.html>`__        |
```
[the remark this answers is not in the document]
Comment (eric-kwok-nt) [reply docs/source/resources/02_bibliography.rst:78]: Made a mistake on this, I have changed it point to v0.11

**Comment by eric-kwok-nt** [comment]
> `"r101-fpn"` should be monospace as well.

Amended this in mask_rcnn.py


> Some additional formatting/consistency issues: [1](https://github.com/aimakerspace/PeekingDuck/blob/node-segment/peekingduck/pipeline/nodes/model/mask_rcnn.py#L29): `"r101-fpn"` should be monospace as well. [2](https://github.com/aimakerspace/PeekingDuck/blob/node-segment/peekingduck/pipeline/nodes/model/mask_rcnn.py#L76): The inference code referenced here does not match the code pointed by the URL in Bibliography. We should probably avoid using the `stable` URL as it will change when torchvision increments its version number.

Amended this in the mask_rcnn.py as well

File: peekingduck/pipeline/nodes/model/mask_rcnn.py [hunk 5]
```diff
@@ -73,8 +73,8 @@ class Node(AbstractNode):  # pylint: disable=too-few-public-methods
         https://arxiv.org/abs/1703.06870
 
         Inference code adapted from:
-        https://pytorch.org/vision/stable/_modules/torchvision/models/detection/mask_rcnn.html
-        Model weights adapted from:
+        https://pytorch.org/vision/0.11/_modules/torchvision/models/detection/mask_rcnn.html
+        The weights for Mask-RCNN Model with ResNet50 FPN backbone were adapted from:
         https://download.pytorch.org/models/maskrcnn_resnet50_fpn_coco-bf2d0c1e.pth
```
Comment (liyier90) [inline peekingduck/pipeline/nodes/model/mask_rcnn.py:78]: Consider adding a vertical whitespace so this is rendered on a new line. 

image
Similar to how "Mask R-CNN:..." and "Inference code ..." are on separate line.

**Review by liyier90 — COMMENTED** [review]
Just one more minor formatting comment.

File: peekingduck/pipeline/nodes/model/mask_rcnn.py (hunk 5 above)
Comment (eric-kwok-nt) [reply peekingduck/pipeline/nodes/model/mask_rcnn.py:78]: amended

**Pull request merged**